services/app-api/src/handlers/apollo_gql.ts

```diff
@@ -81,8 +88,21 @@ function contextForRequestForFetcher(userFetcher: userFromAuthProvider): ({
                 }
 
                 const store = NewPostgresStore(pgResult)
-                const userResult = await userFetcher(authProvider, store)
+                const userId = event.requestContext.authorizer?.principalId
+
+                let userResult
+                if (authProvider && !fromThirdPartyAuthorizer) {
+                    userResult = await userFetcher(authProvider, store)
+                } else if (fromThirdPartyAuthorizer) {
+                    userResult = await userFromThirdPartyAuthorizer(
```

Comment:
I'm confused how this compiles. This function looks like it takes a `string` for userID, but userID with the question mark in the chain should be possibly `undefined`

Reply:
oh duh, `&& userId` right above this line.